controller/volume_controller.go

```diff
@@ -604,9 +604,11 @@ func (vc *VolumeController) cleanupCorruptedOrStaleReplicas(v *longhorn.Volume,
 			// Skip running healthy replicas.
 			// Skip degraded volume.
 			// Skip if eviction is going.
+			// Skip if datalocality has been enabled
 			if r.Spec.HealthyAt != "" ||
 				healthyCount < v.Spec.NumberOfReplicas ||
-				hasEvictionRequestedReplicas {
+				hasEvictionRequestedReplicas ||
+				!isDataLocalityDisabled(v) {
```

Comment:
Should we limit to the replica that exists on the same attached node of v? cc @PhanLe1010